src/main/java/uk/nhs/hee/tis/revalidation/connection/service/MasterElasticSearchService.java

```diff
@@ -38,11 +52,65 @@
   @Autowired
   ConnectionInfoMapper connectionInfoMapper;
 
+  final int PAGE_SIZE = 100;
+
   /**
    * find all trainee from ES Master Index.
    */
   public List<ConnectionInfoDto> findAllMasterToDto() {
     Iterable<MasterDoctorView> masterList = masterElasticSearchRepository.findAll();
     return connectionInfoMapper.masterToDtos(masterList);
   }
+
+  /**
+   * find all trainee from ES Master Index.
+   * (with Scroll API to avoid ES index max_result_window excess error)
+   */
+  @SneakyThrows
+  public List<ConnectionInfoDto> findAllScroll() {
+
+    RestHighLevelClient client = new RestHighLevelClient(
+        RestClient.builder(
+            new HttpHost("localhost", 9200, "http")));
```

Comment:
Just a thought, in stage and prod how this value will  dynamically work?

Reply:
Added dynamic mapping for ES UR, so it can provide correct value for local, stage and prod HttpHost.